### ISSUE TITLE
Advanced keyboard shortcuts

### ISSUE THREAD

**Issue by marktnoonan** [cypress-io/cypress#24973]
### What would you like?

Cypress has a powerful UI with many features. Some workflows could be sped up with keyboard shortcuts.

##### General (inc. both specs lists)

- `Cmd + K` to focus the search bar to search for specs
- `Enter` when searching to "run all" matching specs (when "run all" is enabled) (think @lmiller1990 suggested this?)
- Pressing `down` when on search bar to move to the first item in the result list
- Arrow key behavior in main Spec Explorer should match the behavior of the inline specs list - arrows for navigation between items
- `?` key to open the list of keyboard shortcuts - and corresponding `?` icon, instead of mac `command` icon, in the notes
- list of keyboard shortcuts should explain which contexts the shortcuts work in
- Command palette to quickly navigate to different parts of the app
- A shortcut to open the current spec in vs code

##### In Spec Runner page

- `Cmd + B` to toggle the file list (would not have to drop `f`, but lots of people have muscle memory from vscode, we had this previously & it is probably more likely to be discovered by accident than the `f` hotkey)
- `Cmd + leftarrow`, `Cmd + rightarrow` or similar way to move keyboard focus between primary areas of the page - nav, specs list, command log, AUT header, and AUT itself.

### Why is this needed?

It would make it more pleasant to use Cypress primarily with the keyboard, as well as improve accessibility. It would generally improve the workflow of bouncing back and forth between editing a test in a text editor (a keyboard based task) and interacting with Cypress, which is currently heavily mouse based for certain tasks.

### Other

The suggestions above of `Cmd + {some key}` need to be evaluated in the context of existing browser functionality and screenreader commands, and shouldn't step on known browser behavior. So I would expect some of the actual items in here to change in the implementation.

**Comment by lmiller1990** [comment]
> Command pallet to quickly navigate to different parts of the app

This seems like quite a lot of work (designs, etc).

> A shortcut to open the current spec in vs code

Like this https://github.com/cypress-io/cypress/issues/23228. Someone almost implemented it! We can easily write this plugin but we need to support `--spec` for open mode. We don't have this, I don't know why, but I think there's historical context around this.

The rest seem pretty straight forward and reasonable.

**Comment by marktnoonan** [comment]
> This seems like quite a lot of work (designs, etc).

Yeah it's the biggest one for sure, it puts the epic in this epic. I think it's a good candidate for release + incrementally improve/add capabilities. Like what's the smallest first step could take in a command palette? Jump to spec list, settings page, runs page? Create new spec? Change browser? Individually pretty small steps.

Maybe later we'd add spec search, other things.

At its core it's a combobox in a modal with autocomplete (if design is cool with that).

> A shortcut to open the current spec in vs code

I mean, from Cypress, a shortcut to open the open spec file in VS Code (instead of having to click). But yeah that other one is good too, I know an extension is in-progress.

**Comment by lmiller1990** [comment]
> I mean, from Cypress, a shortcut to open the open spec file in VS Code (instead of having to click). But yeah that other one is good too, I know an extension is in-progress.

Right, gotcha. This would be quite easy, then. I would really like to see the VS Code plugin, too - Playwright has this, it's really useful.

**Comment by VOJAKIS** [comment]
As @jennifer-shehane mentioned the #29035 issue, the go down could be implemented as a shortcut. But it solely depends on the design, that the developers will choose.
I would like to see maybe both implemented, as a shortcut and as a button, as is in the #29035 issue.
